next-netflix/app/Search/page.tsx

```diff
@@ -0,0 +1,189 @@
+  const [isDataLoading, setIsDataLoading] = useState(false); // isLoading 상태 추가
+  const bottom = useRef(null); 
+
+  const fetchSearchData = async ({ queryKey, pageParam = 1 }: { queryKey: [string]; pageParam?: number | undefined }) => {
+    const searchInput = queryKey[0];
+    const apiKey = "4a427cc6585f047c91a2fa3483fb8d31";
+    const res = await fetch(`https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&query=${searchInput}&page=${pageParam}`);
+    const searchData = await res.json() as IMovie[];
+    return searchData;
+  };
+
+  const fetchPopularData = async ({pageParam = 1 }: { pageParam?: number | undefined } ) => {
+    const apiKey = "4a427cc6585f047c91a2fa3483fb8d31";
+    const res = await fetch(`https://api.themoviedb.org/3/movie/popular?api_key=${apiKey}&page=${pageParam}`);
+    const popularData = await res.json() as IMovie[];
+    return popularData;
+  };
+
+  const {
+    data,
+    fetchNextPage,
+    isFetching,
+    isLoading : isQueryLoading,
+  } = useInfiniteQuery(
+    ['searchData', searchInput],
+    ({ pageParam }) => {
+      return searchInput
+      ? fetchSearchData({ queryKey: [searchInput], pageParam })
+      : fetchPopularData({ pageParam });
+    },
+    {
+      getNextPageParam: (lastPage) => {
+        const currentPage = lastPage?.[lastPage.length - 1]?.page || 1;
+        return currentPage + 1;
+      }
+    }
+  );
+
+  const onIntersect = ([entry]: IntersectionObserverEntry[]) => entry.isIntersecting && fetchNextPage()
+
+  // Scroll로 bottom ref와 onIntersect를 넘겨줌
+  Scrolling({
+    target: bottom,
+    onIntersect,
+  })
+
+  useEffect(() => {
+    if (data) {
+      const updatedResults = data.pages.flatMap((page) => page);
+      setSearchResults(updatedResults);
+      setIsDataLoading(false); // 데이터 로드 완료 후 isLoading 값을 false로 설정
+    }
+  }, [data]);
+
+  useEffect(() => {
+    setIsDataLoading(isQueryLoading);
+  }, [isQueryLoading]);
+
+  return (
+    <div className={styles.container}>
+      <SearchBar>
+        <AiOutlineSearch style={{color : "#C4C4C4", width : '30px'}} />
+        <SearchInput 
+        placeholder="Search for a show, movie, genre, e.t.c" 
+        value={searchInput}
+        onChange={(e) => setSearchInput(e.target.value)}/>
+        <GrFormClose style={{color : "red", width : '30px'}} />
+      </SearchBar>
+      
+      <ListWrapper>
+        <TitleText>Top Searches</TitleText>
+        {!isQueryLoading && !isDataLoading && !isFetching && data && data.pages && (
+          <List>
+          {searchResults.map((movie) => {
+            console.log(movie); // movie 로그 출력
+            return movie.results.map((result) => {
+              console.log(result); // result 로그 출력
+              return (
+                <div key={result.id}>
+                  {result.id ? (
+                    <InnerList>
+                      <Img src={`https://image.tmdb.org/t/p/original${result.poster_path}`} alt={result.title} />
```

Comment:
result의 id는 있지만 poster가 없을 수 있어서
```suggestion
   {result.poster_path === null ? ( (
   <Img src="/error.jpg" alt={result.title} />
   ) : (
   <Img src={`https://image.tmdb.org/t/p/original${result.poster_path}`} alt={result.title} />
  )}
```
로 예외처리 해주는 게 좋을 것 같아요!

Reply:
아하 어쩐지 종종 poster_path 불러오는데 엑박이 뜨는 부분들이 있더라구요..! 예외 처리하면 해결이 되겠네요 감사합니다~! Repick 팀에서는 넷플릭스 배경으로 하셨던데 잘 보았습니닷 ㅎㅎ